river/set.zig

```diff
@@ -0,0 +1,51 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2022 The River Developers
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const assert = @import("std").debug.assert;
+
+pub fn Set(comptime T: type, comptime capacity: usize) type {
```

Comment:
This doesn't really need to be generic to be honest, I wouldn't mind calling it KeycodeSet and hardcoding u32 as the element type plus a reasonable capacity.